### DISCUSSION TITLE
FindVariableFeatures() HELP!

### DISCUSSION THREAD

**Discussion by haileye4** [satijalab/seurat#7561] [Q&A]
Hi all! I'm very new to Seurat and scRNA-seq data... I'm following the introductory tutorial "Guided tutorial- 2,700 PBMCs", and I'm getting a weird error I have no idea how to fix.

I've put in my data (barcodes.tsv, matrix, and features.tsv) as a Seurat object beginning with this:

_ecDNA.data <- Read10X(data.dir = "/mypath.../scRNA-seq/hg19/")
ecDNA <- CreateSeuratObject(counts = ecDNA.data, project = "ecDNA")_

I then followed quality control and select cells for further analysis, and normalize the data:
_ecDNA[["percent.mt"]] <- PercentageFeatureSet(ecDNA, pattern = "^MT-")
ecDNA <- subset(test, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & nCount_RNA > 500 & percent.mt < 5)
ecDNA <- NormalizeData(ecDNA)_

The next step is to use the "FindVariableFeatures" function. I followed it exactly as the tutorial:
_ecDNA <- FindVariableFeatures(object = ecDNA, assay = "RNA", selection.method = "vst", nfeatures = 2000)_

Yet CONTINUE to get this error each time:
**_Calculating gene variances
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Calculating feature variances of standardized and clipped values
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Error in match.arg(arg = layer, choices = Layers(object = object, search = FALSE)) : 
  'arg' should be one of “counts”, “data”, “scale.data”_**

How do I fix this!? Please help I appreciate it! I'm struggling a lot with this function and trying to understand how to resolve this error...

**Comment by haileye4** [comment]
Also, this is what my object looks like from my environment. I can see the "counts", "data", and "scale.data" underneath "RNA" under "assays"! I just don't understand how that function is struggling to access it... I'm also not good with layers and understanding what it means. (BTW "test" is the same as my ecDNA object...)

<img width="890" alt="Screen Shot 2023-07-12 at 12 05 11 PM" src="https://github.com/satijalab/seurat/assets/123395871/626913b9-f869-489f-8b06-20044800355d">

**Comment by Xkimberly** [comment]
have you resolved this problem? i met the same one

**Comment by haileye4** [comment]
Hey! No, I never resolved it :(

**Comment by Forsyl** [comment]
I resolved this problem by reinstall Seurat V5: 
remotes::[install_github](https://remotes.r-lib.org/reference/install_github.html)("satijalab/seurat", "seurat5", quiet = TRUE)
